Count Visited Nodes in a Directed Graph/solution.cpp

```diff
@@ -0,0 +1,70 @@
+#include "solution.h"
+std::vector<int> CountVisitedNodes(const std::vector<int>& edges) {
+  std::vector<int> visited_count(edges.size(), -1);  // result
+  absl::flat_hash_set<size_t> path;
```

Comment:
use seem to be using these only inside the loop, I,.e during one iteration. 
may be more convenient to declare inside the loop. 
This may be a bit less efficient in some cases, but more readable,